### PR TITLE
Frannie's Game Level

### PR DESCRIPTION
**Pull request by frich98**
My level's feature is an AI-theme, with randomly sized blocks over 5 rows. Note that I changed the Player Supervisor C# script to handle the random aspect, but it should not affect anyone else's level (unless of course there are naming similarities for variables that we'll need to address). Let me know if you have any issues with this code or the Player Supervisor script.

### CODE REVIEW

File: Assets/Scripts/PlayerSupervisor.cs [hunk 1]
```diff
@@ -8,13 +8,46 @@ public class PlayerSupervisor : MonoBehaviour
...
 
-    private int points = 0;
+    // to indicate block in current scene (in Inspector)
+    // and parent block, which will always be "Blocks" empty object
+    [SerializeField] GameObject block;
+    [SerializeField] GameObject parentBlock;
+
+    // our project uses fixed 4-3 ratio, so these world units will not be changing
+    [SerializeField] float screenWidthWorld = 16f;
+    [SerializeField] float screenHeightWorld = 12f;
```
Comment (matzingc) [inline Assets/Scripts/PlayerSupervisor.cs:20]: Since these are fixed, I think it would make sense to mark them as const and non-serialized fields. Also, I don't think screenHeightWorld is ever used; though it might be good to keep around for later.

File: Assets/Scripts/PlayerSupervisor.cs [hunk 2]
```diff
@@ -8,13 +8,46 @@ public class PlayerSupervisor : MonoBehaviour
...
+
+    // for random block settings - max and min widths
+    [SerializeField] int minBlockWidthWorld = 1;
+    [SerializeField] int maxBlockWidthWorld = 4;
+
+    // for random block settings - start row to place blocks
+    [SerializeField] float blockStartRow = 5f;
```
Comment (matzingc) [inline Assets/Scripts/PlayerSupervisor.cs:27]: It might be good to add a [Range()] decoration on here to control the max/min starting row for the blocks so they don't end up on top of the paddle or above the ceiling.

File: Assets/Scripts/PlayerSupervisor.cs [hunk 3]
```diff
@@ -8,13 +8,46 @@ public class PlayerSupervisor : MonoBehaviour
     [SerializeField] GameManager gameManager;
     [SerializeField] int activeBlocks;
 
-    private int points = 0;
+    // to indicate block in current scene (in Inspector)
+    // and parent block, which will always be "Blocks" empty object
+    [SerializeField] GameObject block;
```
Comment (matzingc) [inline Assets/Scripts/PlayerSupervisor.cs:15]: A more appropriate name for this might be blockType - or maybe just a bool generateRandomBlocks?

File: Assets/Scripts/PlayerSupervisor.cs [hunk 4]
```diff
@@ -8,13 +8,46 @@ public class PlayerSupervisor : MonoBehaviour
...
+    [SerializeField] int minBlockWidthWorld = 1;
+    [SerializeField] int maxBlockWidthWorld = 4;
+
+    // for random block settings - start row to place blocks
+    [SerializeField] float blockStartRow = 5f;
+
+    // starting block width
+    [SerializeField] int prefabBlockWidthWorld = 1;
+
+    [SerializeField] float spaceBetweenBlocks = .05f;
```
Comment (matzingc) [inline Assets/Scripts/PlayerSupervisor.cs:32]: It looks like this is instead saved your scene's Player Supervisor as .1, but the Player Supervisor prefab has it as .05. I'm not sure if anything should be adjusted so they're all consistent, though the .1 looks really nice in your scene.

File: Assets/Scripts/PlayerSupervisor.cs [hunk 5]
```diff
@@ -23,11 +56,129 @@ void Update()
...
+                blockWidth = widths[0]; // block widths in position 0
+                blockMidpoint = calculateBlockMidpoint(blocksPlaced, blockWidth, 
+                                                       priorBlockWidth, priorBlockMidpoint);
+                priorBlockWidth = blockWidth;
+                priorBlockMidpoint = blockMidpoint;
+                GameObject newBlock = createBlock(blockMidpoint, blockRow);
+
+                // update block scale based on block width, note scale is a PERCENTAGE
+                scaleChange = new Vector2(blockWidth/prefabBlockWidthWorld - 1, 0);
+                newBlock.transform.localScale += scaleChange;
```
Comment (matzingc) [inline Assets/Scripts/PlayerSupervisor.cs:169]: This was a little hard to follow - I think mostly because of the tricky math. It might make more sense here to remove the -1 and make the localScale a straight set rather than +=. I think the math here relies on localScale always starting as 1 so the initial value and the -1 seem to be a wash, but I'm sure you understand what's going on here better than I do and trust your judgement if there's a case for using the initial localScale value when resizing. 
```suggestion
                scaleChange = new Vector2(blockWidth/prefabBlockWidthWorld, 0);
                newBlock.transform.localScale = scaleChange;

```

**Comment by frich98** [comment]
Cancelling this merge request as I'll be making a new one to address the merge issue with main. Will add the comments above into the new request.

File: Assets/Scripts/PlayerSupervisor.cs (hunk 1 above)
Comment (frich98) [reply Assets/Scripts/PlayerSupervisor.cs:20]: good call, did this

File: Assets/Scripts/PlayerSupervisor.cs (hunk 2 above)
Comment (frich98) [reply Assets/Scripts/PlayerSupervisor.cs:27]: nice!! love this. updated it in code.

File: Assets/Scripts/PlayerSupervisor.cs (hunk 3 above)
Comment (frich98) [reply Assets/Scripts/PlayerSupervisor.cs:15]: good idea - changed this

File: Assets/Scripts/PlayerSupervisor.cs (hunk 4 above)
Comment (frich98) [reply Assets/Scripts/PlayerSupervisor.cs:32]: changed this - agreed, the 0.1 looks best and matched the code w/ the prefab PS value.

File: Assets/Scripts/PlayerSupervisor.cs (hunk 5 above)
Comment (frich98) [reply Assets/Scripts/PlayerSupervisor.cs:169]: agreed this is confusing, changed to your suggestion!
